vhost/src/vhost_user/gpu_backend_req.rs

```diff
@@ -121,6 +147,12 @@ impl GpuBackend {
         self.send_message(GpuBackendReq::GET_EDID, get_edid, None)
     }
 
+    /// Forward 2d gpu command SetScanout requests to the frontend.
+    pub fn set_scanout(&mut self, scanout_id: &u32) -> io::Result<()> {
+        self.send_message_no_reply(GpuBackendReq::SCANOUT, scanout_id, None)
```

Comment:
You seem to be sending incomplete data. According to the spec and QEMU source code you should be sending this struct:
```C
typedef struct VhostUserGpuScanout {
    uint32_t scanout_id;
    uint32_t width;
    uint32_t height;
} QEMU_PACKED VhostUserGpuScanout;
```

Reply:
mmm, Right!